src/util/string.h

```diff
@@ -101,6 +101,20 @@ bool parseColorString(const std::string &value, video::SColor &color, bool quiet
 		unsigned char default_alpha = 0xff);
 std::string encodeHexColorString(video::SColor color);
 
+/**
+ * Converts a letter to lowercase, with safe handling of the char type and non-ASCII.
+ * @param c input letter
+ * @returns same letter but uppercase
+*/
+inline char my_tolower(char c)
+{
+	// tolower uses the POSIX locale, which will be UTF-8 on any reasonable system
+	// but not on Windows. In that case passing any high codepoints would mangle them.
+	if (static_cast<unsigned char>(c) > 0x7f)
+		return c;
```

Comment:
If the problem is just Windows, couldn't you fix this by using `_tolower_l` and `_create_locale`?

Reply:
Well Windows is just the biggest offender because it has a non-utf-8 locale by default, you can set the same on Unix-like systems. `tolower_l` exists on some libcs to solve that, Windows utf-8 locale is also [relatively new](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setlocale-wsetlocale?view=msvc-170#utf-8-support), then you have inconsistent implementations, support code needed etc...
Not to mention that this is all pointless because if you want to properly lowercase an UTF-8 string you have to decode it first (not operate on `char`).

I guess I should reword the comment.